daemon/openrazer_daemon/hardware/device_base.py

```diff
@@ -104,13 +134,371 @@ def __init__(self, device_path, device_number, config, testing=False, additional
             ('razer.device.misc', 'hasDedicatedMacroKeys', self.dedicated_macro_keys, None, 'b'),
         }
 
+        effect_methods = {
+            ('razer.device.misc', 'getCurrentEffect', self.get_current_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentEffectColors', self.get_current_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentEffectSpeed', self.get_current_effect_speed, None, 'i'),
+            ('razer.device.misc', 'getCurrentWaveDir', self.get_current_wave_dir, None, 'i'),
+        }
+
+        logo_effect_methods = {
+            ('razer.device.misc', 'getCurrentLogoEffect', self.get_current_logo_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentLogoEffectColors', self.get_current_logo_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentLogoEffectSpeed', self.get_current_logo_effect_speed, None, 'i'),
+        }
```

Comment:
Shouldn't the other led locations also have the wave direction method?

Reply:
Done.